src/main/services/CaseService.ts

```diff
@@ -1,83 +1,129 @@
-import axios, { AxiosInstance, AxiosResponse } from 'axios';
+import axiosService, { AxiosInstance, AxiosResponse } from 'axios';
 import config from 'config';
 import FormData from 'form-data';
 
 import { CaseApiDataResponse } from '../definitions/api/caseApiResponse';
 import { DocumentUploadResponse } from '../definitions/api/documentApiResponse';
 import { DocumentDetailsResponse } from '../definitions/api/documentDetailsResponse';
 import { UserDetails } from '../definitions/appRequest';
-import { CaseDataCacheKey, CaseWithId } from '../definitions/case';
+import { CaseWithId } from '../definitions/case';
 import { JavaApiUrls } from '../definitions/constants';
 import { toApiFormat, toApiFormatCreate } from '../helper/ApiFormatter';
 
+import { axiosErrorDetails } from './AxiosErrorAdapter';
+
 export class CaseApi {
-  constructor(private readonly axio: AxiosInstance) {}
+  constructor(private readonly axios: AxiosInstance) {}
 
   createCase = async (caseData: string, userDetails: UserDetails): Promise<AxiosResponse<CaseApiDataResponse>> => {
-    const userDataMap: Map<CaseDataCacheKey, string> = new Map(JSON.parse(caseData));
-    const body = toApiFormatCreate(userDataMap, userDetails);
-    return this.axio.post(JavaApiUrls.INITIATE_CASE_DRAFT, body);
+    try {
+      return await this.axios.post(
+        JavaApiUrls.INITIATE_CASE_DRAFT,
+        toApiFormatCreate(new Map(JSON.parse(caseData)), userDetails)
+      );
+    } catch (error) {
+      throw new Error('Error creating case: ' + axiosErrorDetails(error));
+    }
   };
 
   getUserCases = async (): Promise<AxiosResponse<CaseApiDataResponse[]>> => {
-    return this.axio.get<CaseApiDataResponse[]>(JavaApiUrls.GET_CASES);
+    try {
+      return await this.axios.get<CaseApiDataResponse[]>(JavaApiUrls.GET_CASES);
+    } catch (error) {
+      throw new Error('Error getting user cases: ' + axiosErrorDetails(error));
+    }
   };
 
   downloadClaimPdf = async (caseId: string): Promise<AxiosResponse> => {
-    const data = {
-      caseId,
-    };
-    return this.axio.post(`${JavaApiUrls.DOWNLOAD_CLAIM_PDF}`, data, {
-      responseType: 'arraybuffer',
-      headers: {
-        'Content-Type': 'application/pdf',
-      },
-    });
+    try {
+      return await this.axios.post(
+        `${JavaApiUrls.DOWNLOAD_CLAIM_PDF}`,
+        {
+          caseId,
+        },
+        {
+          responseType: 'arraybuffer',
+          headers: {
+            'Content-Type': 'application/pdf',
+          },
+        }
+      );
+    } catch (error) {
+      throw new Error('Error downloading claim pdf: ' + axiosErrorDetails(error));
+    }
   };
 
   getCaseDocument = async (docId: string): Promise<AxiosResponse> => {
-    return this.axio.get(`${JavaApiUrls.DOCUMENT_DOWNLOAD}${docId}`, {
-      responseType: 'arraybuffer',
-    });
+    try {
+      return await this.axios.get(`${JavaApiUrls.DOCUMENT_DOWNLOAD}${docId}`, {
+        responseType: 'arraybuffer',
+      });
```

Comment:
single line?

Reply:
Would be too long, can do the same here.